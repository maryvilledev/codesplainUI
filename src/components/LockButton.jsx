import React, { PropTypes } from 'react';

import IconButton from 'material-ui/IconButton';
import Lock from 'material-ui/svg-icons/action/lock';
import LockOpen from 'material-ui/svg-icons/action/lock-open';

const style = {
  zIndex: 5,
};

const LockButton = ({ onClick, readOnly }) => {
  const lockIcon = readOnly ? <Lock /> : <LockOpen />;
  const toolTipText = readOnly ? 'Switching back to edit mode not supported yet' : 'Click to lock the snippet (this will prevent further changes.)';
  return (
    <IconButton
      disabled={readOnly}
      label={`Switch to ${readOnly ? 'edit' : 'read-only'} mode`}
      onTouchTap={onClick}
      tooltip={toolTipText}
      style={style}
    >
      {lockIcon}
    </IconButton>
  );
};

LockButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
};

export default LockButton;
