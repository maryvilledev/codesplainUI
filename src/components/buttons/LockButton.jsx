import React, { PropTypes } from 'react';

import IconButton from 'material-ui/IconButton';
import Lock from 'material-ui/svg-icons/action/lock';
import LockOpen from 'material-ui/svg-icons/action/lock-open';

const styles = {
  button: {
    zIndex: 5,
  },
  buttonContainer: {
    height: '48px',
  },
};

const LockButton = ({ id, onClick, readOnly }) => {
  const lockIcon = readOnly ? <Lock /> : <LockOpen />;
  const toolTipText = readOnly ? 'Switching back to edit mode not supported yet' : 'Click to lock the snippet (this will prevent further changes.)';
  return (
    <span style={styles.buttonContainer} title={toolTipText}>
      <IconButton
        id={id}
        disabled={readOnly}
        label={`Switch to ${readOnly ? 'edit' : 'read-only'} mode`}
        onTouchTap={onClick}
        style={styles.button}
      >
        {lockIcon}
      </IconButton>
    </span>
  );
};

LockButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
};

export default LockButton;
