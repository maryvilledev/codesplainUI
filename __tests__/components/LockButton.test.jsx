import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import LockButton from '../../src/components/buttons/LockButton';

describe('LockButton />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot of when the app is not in read-only mode', () => {
    const wrapper = shallowWithContext(
      <LockButton
        onClick={jest.fn()}
        readOnly={false}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('matches snapshot of when the app is in read-only mode', () => {
    const wrapper = shallowWithContext(
      <LockButton
        onClick={jest.fn()}
        readOnly={true}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
})
