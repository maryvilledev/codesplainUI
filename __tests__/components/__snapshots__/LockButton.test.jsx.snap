exports[`<LockButton /> matches snapshot of when the app is in read-only mode 1`] = `
<span
  style={
    Object {
      "height": "48px",
    }
  }
  title="Switching back to edit mode not supported yet">
  <IconButton
    disableTouchRipple={false}
    disabled={true}
    iconStyle={Object {}}
    id="lockButton"
    label="Switch to edit mode"
    onTouchTap={[Function]}
    style={
      Object {
        "zIndex": 5,
      }
    }
    tooltipPosition="bottom-center"
    touch={false}>
    <ActionLock />
  </IconButton>
</span>
`;

exports[`<LockButton /> matches snapshot of when the app is not in read-only mode 1`] = `
<span
  style={
    Object {
      "height": "48px",
    }
  }
  title="Click to lock the snippet (this will prevent further changes.)">
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    id="lockButton"
    label="Switch to read-only mode"
    onTouchTap={[Function]}
    style={
      Object {
        "zIndex": 5,
      }
    }
    tooltipPosition="bottom-center"
    touch={false}>
    <ActionLockOpen />
  </IconButton>
</span>
`;
